src/extensions/default/CodeFolding/foldhelpers/handlebarsFold.js

```diff
@@ -0,0 +1,158 @@
+/**
+ * Fold range finder for handlebars/mustache template type files.
+ * @author Patrick Oladimeji
+ * @license MIT
+ * @date 14/08/2016 22:04:21
+ */
+
+define(function (require, exports, module) {
+    "use strict";
+    var CodeMirror = brackets.getModule("thirdparty/CodeMirror/lib/codemirror"),
+        space = /\s/;
+    /**
+     * Utility function for scanning the text in a document until a certain condition is met
+     * @param {object}   cm        The code mirror object representing the document
+     * @param {string}   startCh   The start character position for the scan operation
+     * @param {number}   startLine The start line position for the scan operation
+     * @param {function} condition A predicate function that takes in the text seen so far
+     */
+    function scanTextUntil(cm, startCh, startLine, condition) {
+        var line = cm.getLine(startLine);
+        var seen = '',
+            characterIndex = startCh,
+            currentLine = startLine;
+        while (currentLine <= cm.lastLine()) {
+            if (line.length === 0) {
+                characterIndex = 0;
+                line = cm.getLine(++currentLine);
+            } else {
+                seen = seen.concat(line[characterIndex] || '');
+                if (condition(seen)) {
+                    return {from: {ch: startCh, line: startLine}, to: {ch: characterIndex, line: currentLine}, string: seen};
```

Comment:
Could you hoist this in a variable and use some newlines?
At least something like:
```js
var obj = {
  from: {ch: startCh, line: startLine},
  to: {ch: characterIndex, line: currentLine},
  string: seen
};
return obj;
```
Better name welcome.
It will make diff easier to read.